imports/api/listings/server/methods.js

```
@@ -0,0 +1,199 @@
+            {
+                $sort: {count: -1} // sort the groups on count field in descending order
+            },
+            {
+                $limit: 1 // get the top-most document
+            }
+        ]).toArray()
+
+        if (creatorOfTheDay[0] != null && creatorOfTheDay[0] != undefined){
+            var creatorUsername = getUserNameInfo(creatorOfTheDay[0]._id)
+            creatorOfTheDay[0]["creatorUsername"] = creatorUsername?.username?.value
+        }
+
+        return creatorOfTheDay
+    }
+});
+
+//getFormattedDate to get date in format (2022-04-12)
+function getFormattedDate(date){
+
+    var monthString = (date.getMonth() + 1) + ""
+    if (monthString.length == 1){
+        monthString = "0" + (date.getMonth() + 1)
+    }
+
+    var dateString = date.getDate() + ""
+    if (dateString.length == 1){
+        dateString = "0" + date.getDate()
+    }
+
+    var formattedDate = date.getFullYear() + "-" + monthString + "-" + dateString
+    return formattedDate
+}
+
+//getting the nft url out of the recipe object
+function getNftUrl(recipe) {
+
+    var nftUrl = ""
+    var itemOutputs = recipe?.entries?.itemOutputs
+    if (itemOutputs != null && itemOutputs != undefined ){
+        if (itemOutputs[0] != null){
+            var properties = itemOutputs[0].strings
+            for (var i = 0; i < properties.length; i++){
+                if (properties[i].key == "NFT_URL"){
+                    nftUrl = properties[i].value
+                    break;
+                }
+            }
+            
+        }
+    }
+    return nftUrl
+}
+
+//getting the nft name out of the recipe object
+function getNftName(recipe) {
+
+    return recipe?.name
+}
+
+//fetching username info 
+function getUserNameInfo(address){
+    var result;
+    try{
+        let response = HTTP.get(
+            sanitizeUrl(`${Meteor.settings.remote.api}/pylons/account/address/${address}`)
+        );
+        result = JSON.parse(response.content);
+    }
+    catch(e){
```

Same empty catch error block here.